e2e/start-companion-with-load-balancer.mjs

```diff
@@ -45,20 +46,38 @@ function createLoadBalancer (baseUrls) {
   return server
 }
 
-const startCompanion = ({ name, port }) => execa('nodemon', [
-  '--watch', 'packages/@uppy/companion/src', '--exec', 'node', '-r', 'dotenv/config', './packages/@uppy/companion/src/standalone/start-server.js',
-], {
-  cwd: new URL('../', import.meta.url),
-  stdio: 'inherit',
-  env: {
-    // Note: these env variables will override anything set in .env
-    COMPANION_PORT: port,
-    COMPANION_SECRET: 'development', // multi instance will not work without secret set
-    COMPANION_PREAUTH_SECRET: 'development', // multi instance will not work without secret set
-    COMPANION_ALLOW_LOCAL_URLS: 'true',
-    COMPANION_LOGGER_PROCESS_NAME: name,
-  },
-})
+const isWindows = process.platform === 'win32'
+const isOSX = process.platform === 'darwin'
+
+const startCompanion = ({ name, port }) => {
+  const cp = spawn(process.execPath, [
+    '-r', 'dotenv/config', ...(isWindows || isOSX ? ['--watch-path', 'packages/@uppy/companion/src', '--watch'] : []), './packages/@uppy/companion/src/standalone/start-server.js',
+  ], {
+    cwd: new URL('../', import.meta.url),
+    stdio: 'inherit',
+    env: {
+      // Note: these env variables will override anything set in .env
+      ...process.env,
+      COMPANION_PORT: port,
+      COMPANION_SECRET: 'development', // multi instance will not work without secret set
+      COMPANION_PREAUTH_SECRET: 'development', // multi instance will not work without secret set
+      COMPANION_ALLOW_LOCAL_URLS: 'true',
+      COMPANION_LOGGER_PROCESS_NAME: name,
+    },
+  })
+  return Object.defineProperty(cp, 'then', {
+    __proto__: null,
+    writable: true,
+    configurable: true,
+    value: Promise.prototype.then.bind(new Promise((resolve, reject) => {
+      cp.on('exit', (code) => {
+        if (code === 0) resolve(cp)
+        else reject(new Error(`Non-zero exit code: ${code}`))
+      })
+      cp.on('error', reject)
+    })),
+  })
+}
```

Comment:
wat dis voodoo

Reply:
We're adding `then` property to the `cp` object so it's awaitable. I'm happy to add a comment, but open to suggestions because I wouldn't know what to put in it / what is the part that needs clarification.

Reply:
what's wrong with the old code? much shorter, easier to read and it's already awaitable, afaik?

Reply:
The old code was using `execa` and `nodemon`; the new one is plain `node`. Nothing's "wrong" with the old code, but I do prefer the new one because it makes us less reliant on external dependencies.

Reply:
I like no extra dependencies, but I agree this is very hard to understand. Comment could say why we are doing this (no extra dep) and just explaining exactly that: We're adding then property to the cp object so it's awaitable.

Reply:
How often will you be reading this code? Don't you think you'll waste more time downloading execa over and over than reading a slightly more complicated but commented code?